### PR TITLE
Adds integration test framework for e2e tests

### PR DESCRIPTION
**Pull request by bharathkkb**
- Adds test setup
- Adds fixtures for following stages
    - bootstrap
    - org
    - env

Todo:

- [x]  CB yaml

- [x] CB Trigger https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/pull/734

### CODE REVIEW

**Comment by rjerrems** [comment]
Overall structure is looking good from my point of view @bharathkkb

File: 1-org/org_policy.tf [hunk 1]
```diff
@@ -149,9 +149,13 @@ module "org_enforce_bucket_level_access" {
...
+  byte_length = 4
+}
+
 module "access_context_manager_policy" {
   source      = "terraform-google-modules/vpc-service-controls/google"
   version     = "~> 2.0.0"
-  parent_id   = local.organization_id
-  policy_name = "default policy"
+  parent_id   = var.org_id
+  policy_name = "default policy-${random_id.suffix.hex}"
```
Comment (bharathkkb) [inline 1-org/org_policy.tf:160]: @rjerrems I am having some test failures [1] for this `access_context_manager_policy` resource. Looks like in one of the previous runs [2] a destroy step failed leaving this behind. I initially thought  I could bypass with a rand suffix, but it doesnt seem dependent on the policy. Any ideas on how we can prevent this in the future or clean up? 

[1]: https://pantheon.corp.google.com/cloud-build/builds/9cfa6a60-7055-4826-b5f7-57fd5fd26b0e;step=5?project=cloud-foundation-cicd
[2]: https://pantheon.corp.google.com/cloud-build/builds/fd9ecc79-f5aa-4c72-bc82-1f6ce6fb095f?project=cloud-foundation-cicd

Comment (rjerrems) [reply 1-org/org_policy.tf:160]: Yeah I think we need to make sure this resource is cleaned up, even if it requires some gcloud commands. An org can only have one access context manager policy from my testing with gcloud commans

File: 0-bootstrap/main.tf [hunk 2]
```diff
@@ -43,15 +43,18 @@ resource "google_folder" "seed" {
 }
 
 module "seed_bootstrap" {
-  source                  = "terraform-google-modules/bootstrap/google"
-  version                 = "~> 1.0"
+  source = "github.com/terraform-google-modules/terraform-google-bootstrap"
+  # version                 = "~> 1.0"
```
Comment (rjerrems) [inline 0-bootstrap/main.tf:47]: is this supposed to be commented out?

Comment (bharathkkb) [reply 0-bootstrap/main.tf:47]: @rjerrems this was temporary as I was waiting for the new 1.2 bootstrap release, will revert.

File: 1-org/org_policy.tf (hunk 1 above)
Comment (bharathkkb) [reply 1-org/org_policy.tf:160]: gotcha, sounds good.

**Comment by bharathkkb** [comment]
@rjerrems this should be ready for review, I will add some kitchen tests in a subsequent PR.

File: 0-bootstrap/main.tf (hunk 2 above)
Comment (rjerrems) [reply 0-bootstrap/main.tf:47]: It looks like this is still commented out?

File: 1-org/org_policy.tf (hunk 1 above)
Comment (rjerrems) [inline 1-org/org_policy.tf:160]: nit: the random suffice is not really required for this resource given that only one can exist in an org

File: build/int.cloudbuild.yaml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2020 Google LLC
```
Comment (rjerrems) [inline build/int.cloudbuild.yaml:1]: nit: this should be 2020

File: test/fixtures/org/main.tf [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ * limitations under the License.
+ */
+
+module "test" {
+  source                    = "../../../1-org"
+  parent_folder             = var.parent_folder
+  org_id                    = var.org_id
+  billing_account           = var.billing_account
+  terraform_service_account = var.terraform_sa_email
+  default_region            = "us-east4"
```
Comment (rjerrems) [inline test/fixtures/org/main.tf:23]: nit: it would be good to have some consistency with the regions used

File: build/int.cloudbuild.yaml (hunk 3 above)
Comment (rjerrems) [reply build/int.cloudbuild.yaml:1]: There are a few instances of this FYI

File: 0-bootstrap/main.tf (hunk 2 above)
Comment (bharathkkb) [reply 0-bootstrap/main.tf]: just fixed :)

File: 1-org/org_policy.tf (hunk 1 above)
Comment (bharathkkb) [reply 1-org/org_policy.tf]: removed

File: build/int.cloudbuild.yaml (hunk 3 above)
Comment (bharathkkb) [reply build/int.cloudbuild.yaml]: done

File: test/fixtures/org/main.tf (hunk 4 above)
Comment (bharathkkb) [reply test/fixtures/org/main.tf]: using `us-central1` for consistency with other fixtures

**Comment by bharathkkb** [comment]
@rjerrems I have addressed the comments but tests are failing due to a [Github packages outage](https://www.githubstatus.com/).  I will re run them when its back.

**Comment by rjerrems** [comment]
I just triggered the builds again, looks like there may be some errors based on the region for BigQuery - I don't think you can specify us-central1

**Comment by bharathkkb** [comment]
@rjerrems issue was here https://github.com/terraform-google-modules/terraform-example-foundation/pull/65#discussion_r453456499 thats probably why i went with `"us-east4"`. Reverting to that now.

**Pull request merged**